src/sage/structure/sage_object.pyx

```diff
@@ -110,15 +110,15 @@ cdef class SageObject:
            a lot of memory.
 
            To support them for a specific class, add a
-           ``cdef public __custom_name`` attribute.
+           ``cdef public _custom_name`` attribute.
         """
```

Comment:
I give up. I was going to say that `__custom_name` indeed looks like it should be private to `class SageObject`, so it should be kept double underscore. However, this encapsulation has been perforated all over the place (cython fault since it didn't mangle).

My preference would be to rename `__custom_name` to `_SageObject__custom_name`, that is explicitly mangling (so it works the same in cython 0.29 and cython 3, with a note to change back to `__custom_name` once cython <3 support is dropped.

Then every use of this private attribute in a different class would be explicit. The only ok usage would be to add `cdef public _SageObject__custom_name` attribute to support renaming of C-extension types. Everything else should be deprecated IMO (I suggested a few places where it is a simple matter of using `rename(..)`.

But of course, all of this is silly, because you should not be responsible of all this technical debt. Whatever, maybe the easier way is really to rename everything to `_SageObject__custom_name` and let the future fix this in due time.